tests/validation/tests/v3_api/test_node.py

```diff
@@ -1,54 +1,1072 @@
+def test_node_label_node_template_delete():
+    client = cluster_detail["client"]
+    cluster = cluster_node_template["cluster"]
+    node_template = cluster_node_template["node_template"]
+    do_cloud_credential = cluster_node_template["do_cloud_credential"]
+    test_label = cluster_node_template["test_label"]
+    create_kubeconfig(cluster_node_template["cluster"])
+
+    nodes = client.list_node(clusterId=cluster.id).data
+
+    existing_labels = {}
+    for node in nodes:
+        existing_labels[node.nodeName] = {}
+        existing_labels[node.nodeName] = node.labels.data_dict()
+
+    # delete label in node template
+    template_label = node_template.labels.data_dict()
+    del template_label[test_label]
+
+    # update node template
+    node_template_new = client.update(node_template, labels=template_label,
+                                      cloudCredentialId=do_cloud_credential.id,
+                                      digitaloceanConfig=
+                                      {"region": "nyc3",
+                                       "size": "2gb",
+                                       "image": "ubuntu-16-04-x64"})
+    node_template_new = client.wait_success(node_template_new)
+    assert test_label not in node_template_new["labels"], \
+        "Label is available on the node template"
+
+    # Add a node in cluster with new node template
+    cluster, node_pools = add_node_cluster(node_template_new, cluster)
+
+    nodes = client.list_node(clusterId=cluster.id).data
+
+    # validate labels on nodes
+    for node in nodes:
+        if node.nodeName not in existing_labels.keys():
+            node_labels = node.labels.data_dict()
+            assert test_label not in node_labels, \
+                "Label is NOT deleted on the node"
+        else:
+            # check if the labels on the existing nodes are intact
+            existing_labels[node.nodeName] == node.labels.data_dict()
+
+
+def test_node_label_node_template_edit_api():
+    test_label = random_name()
+    label_value = random_name()
+    cluster, node_pools, node_template, do_cloud_credential = \
+        create_cluster_node_template_label(test_label, label_value)
+    remove_resource(cluster)
+    client = get_user_client()
+    create_kubeconfig(cluster)
+
+    node = client.list_node(clusterId=cluster.id).data
+    node_id = node[0].id
+    node = client.by_id_node(node_id)
+    # Edit label on node via API
+    node_labels = node.labels.data_dict()
+    assert node_labels[test_label] == label_value
+
+    # edit label through API
+    new_value = random_name()
+    node_labels[test_label] = new_value
+    client.update(node, labels=node_labels)
+    node = client.reload(node)
+    time.sleep(2)
+
+    # Label should be added
+    wait_for_condition(client, node, check_label_added(test_label), None, 10)
+    validate_label_set_on_node(client, node, test_label, new_value)
+
+    remove_resource(cluster)
```

Comment:
you need to pass remove_resource as a parameter to the function in order to use it

Reply:
Added